packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -0,0 +1,242 @@
+import * as Cardano from '../Cardano';
+import { AssetId } from '../Cardano';
+import {
+  AssetsMintedInspection,
+  Inspector,
+  ResolutionResult,
+  assetsBurnedInspector,
+  assetsMintedInspector,
+  resolveInputs,
+  totalAddressInputsValueInspector,
+  totalAddressOutputsValueInspector
+} from './txInspector';
+import { BigIntMath } from '@cardano-sdk/util';
+import { addTokenMaps, subtractTokenMaps } from '../Asset/util';
+import { coalesceValueQuantities } from './coalesceValueQuantities';
+import { subtractValueQuantities } from './subtractValueQuantities';
+
+interface TransactionSummaryInspectorArgs {
+  addresses: Cardano.PaymentAddress[];
+  rewardAccounts: Cardano.RewardAccount[];
+  inputResolver: Cardano.InputResolver;
+  protocolParameters: Cardano.ProtocolParameters;
+}
+
+export type TransactionSummaryInspection = {
+  tokens: Cardano.TokenMap;
+  coins: Cardano.Lovelace;
+  collateral: Cardano.Lovelace;
+  deposit: Cardano.Lovelace;
+  returnDeposit: Cardano.Lovelace;
+  totalFees: Cardano.Lovelace;
+  unresolved: {
+    inputs: Cardano.TxIn[];
+    value: Cardano.Value;
+  };
+};
+
+export type TransactionSummaryInspector = (
+  args: TransactionSummaryInspectorArgs
+) => Inspector<TransactionSummaryInspection>;
```

Comment:
Inspector currently takes in a `HydratedTx` which is a problem, because we want to use these new inspectors for tx confirmation. Can we change it to `Tx`? I think it's also assignable to `HydratedTx` so shouldn't break anything, but also some other inspectors might need `InputResolver` argument added.

Reply:
this PR already makes this change @mkazlauskas , you can see it here in this commit:

https://github.com/input-output-hk/cardano-js-sdk/pull/1057/commits/cff6ef1dbca1c57c07ac102aba9a429fc1ffac2f

But I see I forgot to add it as a breaking change to the commit, message will update it